addons/mail/models/discuss/discuss_channel.py

```diff
@@ -571,14 +571,8 @@ def _notify_thread(self, message, msg_vals=False, **kwargs):
         message_format = message.message_format()[0]
         if "temporary_id" in self.env.context:
             message_format["temporary_id"] = self.env.context["temporary_id"]
-        # Last interest and is_pinned are updated for a channel when posting a message.
-        # So a notification is needed to update UI, and it should come before the
-        # notification of the message itself to ensure the channel automatically opens.
         payload = {"id": self.id, "last_interest_dt": fields.Datetime.now()}
         bus_notifications = [
-            ((self, "members"), "mail.record/insert", {
-                "Thread": {"id": self.id, "is_pinned": True, "model": "discuss.channel"}
-            }),
```

Comment:
I'm not sure why you remove these lines, but they should probably be kept.

The are a lot of issues in tests with these changes.

Reply:
The principle behind the change is moving the decision of pinning the thread from the server to the client.
This is to ensure that the thread is open only when the client actually has new messages on that thread.

That said this change might be a little too drastic as many other flows rely on the thread being pinned by the server. Either this or a lot of tests would need to be moved to js since they now rely on logic client side.

Reply:
There is a big difference between doing it in the server and the client.

When the user is offline, with server it means the channel will be pinned the next time the user opens a page.

If it's done on the client side, it means the channel will never be pinned and the message never seen, which is definitely not what we want.

Reply:
It will be pinned since once the user opens a page they will receive bus notifications about new messages (and deleted messages), thus deciding whether the channel needs to be pinned or not.

Reply:
This is not guaranteed, old bus notifications are eventually discarded in several situations even if they have never been fetched.

Reply:
I see, then I'll try going in the direction of unpinning the channel when a message is deleted (under certain circumstances of course).

Reply:
Currently if I send a message and then delete it, the user receiving the (now deleted) message will see the thread as pinned, even if it has no new messages

Reply:
I see the problem but as I said it's unsolvable. We don't know if the channel was pinned due to the new message (now deleted) or because it was already pinned.

And even if it was pinned from the new message (now deleted), the user might want to keep the channel pinned for whatever reason. It's also possible the user actually saw the message before it got deleted.

Reply:
The specific case I'm referring to (and the task is referring to) is the case of the message being sent and deleted while the receiver is not signed in.
Spec: `No opening of the conversation for the recipient when he signs in `.

In this case the pinning of the channel could have only been an effect of the message post and the receiver would have had no occasion to read the message (except maybe through push notifications). Upon signing in they just see a random channel being pinned with no unread counter and no new messages.